custom_components/foxess_modbus/config_flow.py

```diff
@@ -212,6 +224,35 @@ async def _autodetect_modbus(self, inv_type, host, slave):
             self._errors["base"] = "modbus_error"
         return False, None
 
+    async def _setup_energy_dashboard(self):
+        """Setup Energy Dashboard"""
+        manager = await data.async_get_manager(self.hass)
+        energy_prefs = EnergyPreferencesUpdate(
+            energy_sources=[
+                SolarSourceType(
+                    type="solar", stat_energy_from="sensor.solar_sum_total"
+                ),
+                GridSourceType(
+                    type="grid",
+                    flow_from=[
+                        FlowFromGridSourceType(
+                            stat_energy_from="sensor.grid_consumption_sum_total"
```

Comment:
Currently it's also just available for H1 as well -- it's probably on AC1 etc, but I don't have one to test. It might be on LAN as well for all I know...

If we put in a reimann sum for this on the platforms which don't have a register for now, will it break everything if we later remove that and replace it with a proper register later?

Reply:
Actually, at the least it would either throw away all history or cause a step-change, I think, neither of which is great.